app/group.go

```diff
@@ -372,3 +396,26 @@ func (a *App) UserIsInAdminRoleGroup(userID, syncableID string, syncableType mod
 
 	return true, nil
 }
+
+// GetAllGroupsByTeam iterates all pages of groups associated to the given team and returns them all.
+func (a *App) GetAllGroupsByTeam(teamID string) ([]*model.Group, *model.AppError) {
+	batchCount := 1
+	page := 0
+	var teamGroups []*model.Group
+
+	for batchCount > 0 {
```

Comment:
if this is an internal call (not a public facing api), why does it need to go through pages? wouldn't it make sense to ask for all the groups directly to the database?

also, glancing throught the code, couldn't that be acomplished by passing nil as the `pageOpts` field? (I might have misunderstood the code for it)